pkg/server/admin.go

```diff
@@ -4219,33 +4222,42 @@ func (s *adminServer) RecoveryVerify(
 	return nil, errors.AssertionFailedf("To be implemented by #93043")
 }
 
-// ListTenants returns a list of active tenants in the cluster. Calling this
-// function will start in-process tenants if they are not already running.
+// ListTenants returns a list of tenants that are served
+// by shared-process services in this server.
 func (s *systemAdminServer) ListTenants(
 	ctx context.Context, _ *serverpb.ListTenantsRequest,
 ) (*serverpb.ListTenantsResponse, error) {
-	ie := s.internalExecutor
-	rowIter, err := ie.QueryIterator(ctx, "list-tenants", nil, /* txn */
-		`SELECT name FROM system.tenants WHERE active = true AND name IS NOT NULL`)
-	if err != nil {
-		return nil, err
-	}
-	defer func() { _ = rowIter.Close() }()
-
 	var tenantNames []roachpb.TenantName
-	var hasNext bool
-	for hasNext, err = rowIter.Next(ctx); hasNext && err == nil; hasNext, err = rowIter.Next(ctx) {
-		row := rowIter.Cur()
-		tenantName := tree.MustBeDString(row[0])
-		tenantNames = append(tenantNames, roachpb.TenantName(tenantName))
+	if !s.server.cfg.Settings.Version.IsActive(ctx, clusterversion.V23_1TenantNamesStateAndServiceMode) {
+		tenantNames = append(tenantNames, catconstants.SystemTenantName)
+	} else {
+		ie := s.internalExecutor
+		rowIter, err := ie.QueryIterator(ctx, "list-tenants", nil, /* txn */
+			`SELECT name FROM system.tenants WHERE service_mode = $1 AND name IS NOT NULL`, mtinfopb.ServiceModeShared)
+		if err != nil {
+			return nil, err
+		}
+		defer func() { _ = rowIter.Close() }()
+
+		var hasNext bool
+		for hasNext, err = rowIter.Next(ctx); hasNext && err == nil; hasNext, err = rowIter.Next(ctx) {
+			row := rowIter.Cur()
+			tenantName := tree.MustBeDString(row[0])
+			tenantNames = append(tenantNames, roachpb.TenantName(tenantName))
+		}
 	}
 
 	var tenantList []*serverpb.Tenant
```

Comment:
I suppose we could pre-allocate this since in the common case the tenantList will be the same length as tenantNames.